.github/workflows/pull_request.yml

```diff
@@ -86,6 +89,14 @@ jobs:
       - uses: actions/setup-go@v3
         with:
           go-version: '1.18'
+          cache: true
+      - run: go mod download -x
+      - run: go build -debug-actiongraph=compile.json .
+      - uses: actions/upload-artifact@v3
+        with:
+          name: build-log
+          path: compile.json
```

Comment:
do we still need this?

Reply:
I think it might be interesting to leave it for a bit to see how it changes with cache hits. I doesn't seem to slow things down much. We can remove it anytime if we are no longer interested in the timing.